pkg/artifactory/resource_artifactory_virtual_repository.go

```diff
@@ -199,6 +209,11 @@ func resourceVirtualRepositoryExists(d *schema.ResourceData, m interface{}) (boo
 	key := d.Id()
 	_, resp, err := c.V1.Repositories.GetVirtual(context.Background(), key)
 
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.